h/hamcrest.go

```diff
@@ -873,6 +873,35 @@ func Zero() *Matcher {
 	return m
 }
 
+// FloatNear checks if the value is in range [want - maxDelta, want + maxDelta].
+//
+// Example:
+//   expect.That(t, 100.001, h.FloatNear(100, 0.1))
+func FloatNear(want, maxDelta float64) *Matcher {
+	if maxDelta < 0 {
+		panic("maxDelta < 0")
+	}
+	m := &Matcher{
+		Msg:    fmt.Sprintf("float is near %f within delta %f", want, maxDelta),
+		NotMsg: fmt.Sprintf("float is not near %f within delta %f", want, maxDelta),
+	}
+	m.Match = func(got interface{}) Result {
+		v, ok := got.(float64)
+		if !ok {
+			v2, ok := got.(float32)
+			if !ok {
+				return NewErrorf(got, "expect float")
+			}
+			v = float64(v2)
+		}
```

Comment:
```suggestion
		var v float64
		switch got := got.(type) {
		case float64:
			v = got
		case float32:
			v = float64(got)
		default:
			return NewErrorf(got, "expect float")
		}
```
Optional: I think the type switch is more clear (no extra nesting; no second identifier).

Reply:
Thanks, done.